### PR TITLE
Soumyadeep-best practices

### PR DESCRIPTION
**Pull request by SoumyadeepTrilogy**

### CODE REVIEW

File: restapi/views.py [hunk 1]
```diff
@@ -1,39 +1,60 @@
...
+logger.setLevel(logging.DEBUG)
+
+formatter=logging.Formatter('%(asctime)s:%(levelname)s:%(message)s')
+
+file_handler=logging.FileHandler('views.log')
+file_handler.setLevel(logging.DEBUG)
+file_handler.setFormatter(formatter)
+
+logger.addHandler(file_handler)
+
```
Comment (kushal-ti) [inline restapi/views.py:18]: logging configuration should go into it's separate utility file as a reusable callable method or loggerfactory class

Comment (kushal-ti) [reply restapi/views.py:18]: django specifcally has even better ways to handle this, that you can look into, but from a general perspective, keep your logging configs separate from application logic

File: restapi/views.py [hunk 2]
```diff
@@ -72,6 +98,7 @@ def normalize(expense):
             start += 1
         else:
             end -= 1
+    logger.info("Normalization of expense executed")
     return balances
```
Comment (kushal-ti) [inline restapi/views.py:104]: Use proper naming conventions for classes

File: restapi/views.py [hunk 3]
```diff
@@ -122,19 +163,29 @@ def members(self, request, pk=None):
             for user_id in removed_ids:
                 group.members.remove(user_id)
         group.save()
+        logger.info("logger function executed")
```
Comment (kushal-ti) [inline restapi/views.py:166]: logger function executed ?

File: restapi/views.py [hunk 4]
```diff
@@ -122,19 +163,29 @@ def members(self, request, pk=None):
...
         return Response(status=204)
 
     @action(methods=['get'], detail=True)
-    def expenses(self, _request, pk=None):
+    def expenses(self, _request, pk=None) -> Response:
+        '''
+            Group Expenses Returned
+        '''
+        logger.info("expenses function initiated")
```
Comment (kushal-ti) [inline restapi/views.py:174]: logging could be a lot better. Try adding some contextual data and perhaps some info about time taken to execute a function

File: restapi/views.py [hunk 5]
```diff
@@ -263,10 +338,12 @@ def multiThreadedReader(urls, num_threads):
     """
         Read multiple files through HTTP
     """
+    logger.info("multithreadreader function initiated")
```
Comment (kushal-ti) [inline restapi/views.py:341]: method can be optimized. Recall the section on performance and efficiency

**Review by kushal-ti — CHANGES_REQUESTED** [review]

File: restapi/views.py (hunk 2 above)
Comment (SoumyadeepTrilogy) [reply restapi/views.py:96]: Resolved!

File: restapi/views.py (hunk 1 above)
Comment (SoumyadeepTrilogy) [reply restapi/views.py:10]: Resolved!

File: restapi/views.py (hunk 5 above)
Comment (SoumyadeepTrilogy) [reply restapi/views.py:333]: Done!

File: restapi/views.py (hunk 4 above)
Comment (SoumyadeepTrilogy) [reply restapi/views.py:168]: Resolved it!

**Pull request closed without merging**